src/components/CustomerQuota/NoQuota/NoQuotaCard.test.ts

```diff
@@ -0,0 +1,215 @@
+import { Policy, PastTransactionsResult } from "../../../types";
+import {
+  groupTransactionsByCategory,
+  sortTransactions,
+  TransactionsByCategoryMap
+} from "./NoQuotaCard";
+
+describe("NoQuotaCard utility functions", () => {
+  let sortedTransactions: PastTransactionsResult["pastTransactions"];
+  let allProducts: Policy[];
+  let mockTransactionsByCategoryMap: TransactionsByCategoryMap;
+
+  const latestTransactionTimeMs = 1596530350000;
+  const latestTransactionTime = toDate(latestTransactionTimeMs);
+
+  beforeAll(() => {
+    sortedTransactions = [
+      {
+        category: "tt-token",
+        quantity: 1,
+        identifierInputs: [
+          {
+            ...defaultIdentifier,
+            label: "Device code",
+            value: "AAA987654321"
+          }
+        ],
+        transactionTime: latestTransactionTime
+      },
+      {
+        category: "meal-credits",
+        quantity: 10,
+        transactionTime: latestTransactionTime
+      },
+      {
+        category: "vouchers",
+        quantity: 1,
+        identifierInputs: [
+          {
+            ...defaultIdentifier,
+            label: "Voucher code",
+            value: "AAA987654322"
+          }
+        ],
+        transactionTime: toDate(latestTransactionTimeMs - 10000)
+      },
+      {
+        category: "meal-credits",
+        quantity: 5,
+        transactionTime: toDate(latestTransactionTimeMs - 20000)
+      }
+    ];
+    allProducts = [
+      {
+        category: "meal-credits",
+        name: "Meal credits",
+        order: 0,
+        quantity: { period: 1, limit: 20, unit: { type: "PREFIX", label: "$" } }
+      },
+      {
+        category: "tt-token",
+        name: "TT token",
+        order: 1,
+        quantity: { period: 1, limit: 1 },
+        identifiers: [
+          {
+            label: "Device code",
+            textInput: { disabled: true, visible: true, type: "STRING" },
+            scanButton: { disabled: false, visible: true, type: "QR" }
+          }
+        ]
+      },
+      {
+        category: "vouchers",
+        name: "CDC Vouchers",
+        order: 2,
+        quantity: {
+          period: 1,
+          limit: 1,
+          unit: { type: "POSTFIX", label: " book" }
+        },
+        identifiers: [
+          {
+            label: "Voucher code",
+            textInput: { disabled: false, visible: true, type: "STRING" },
+            scanButton: { disabled: false, visible: true, type: "BARCODE" }
+          }
+        ]
+      }
+    ];
+
+    mockTransactionsByCategoryMap = {
+      "TT token": {
+        transactions: [
+          {
+            transactionDate: "4 Aug 2020, 4:39PM",
+            details: "AAA987654321",
+            quantity: "1 qty",
+            isAppeal: false
+          }
+        ],
+        hasLatestTransaction: true,
+        order: 1
+      },
+      "Meal credits": {
+        transactions: [
+          {
+            transactionDate: "4 Aug 2020, 4:39PM",
```

Comment:
@sebastianquek Hmm looks like the tests are failing because we are formatting the epoch time in UTC +0, but locally it's passing for me (formatted in +8). It also used to pass in the CI (was formatted to +8). I am not familiar with how the timezone is being set (environment variables?) Did something change?

Also I checked in the app by running a local server, it also displays +8 leh

Reply:
Spoke with seb, it's a locale issue -- on local it is +8 and on CI it is +0. Working on a fix now for the CI locale to be configured properly

Reply:
Fixed :D